src/Codegen/Constraints/BaseBuilder.php

```diff
@@ -92,6 +93,64 @@ protected function addEnumConstraintCheck(HackBuilder $hb): void {
     }
   }
 
+  protected function addHackEnumConstraintCheck(HackBuilder $hb): void {
+    $schema = type_assert_type($this->typed_schema, TSchema::class);
+    if (Shapes::keyExists($schema, 'hackEnum')) {
```

Comment:
How about bailing early if the key doesn't exist?
```
if (!Shapes::keyExists($schema, 'hackEnum')) {
    return;
}
```

This way we can un-indent the entire method.